indexing-service/src/main/java/org/apache/druid/indexing/overlord/TaskQueue.java

```diff
@@ -315,6 +320,16 @@ public String apply(Task task)
     }
   }
 
+  private boolean isTaskPending(Task task)
+  {
+    for (TaskRunnerWorkItem workItem : taskRunner.getPendingTasks()) {
+      if (workItem.getTaskId().equals(task.getId())) {
+        return true;
+      }
+    }
+    return false;
```

Comment:
`task.getId()` shouldn't be called by each for loop, and it would be better to change this method to a lambda style.

```java
final String taskId = task.getId();
return taskRunner.getPendingTasks()
                 .stream()
                 .anyMatch(t -> taskId.equals(t.getTaskId()));
```